Version Refactor

Cuphat opened the pull request:
Reworks version checks to make it easier for branches to have their own individual version tracking. Version checks are performed using the branch URL specified, and now checks the supplementary version as well.

The main version bytes at 0x35-0x37 of the ROM are unchanged. A 5-byte set of version bytes are injected to 0x19-0x1D, with branch identifier as the 4th byte and supplementary version as the 5th byte.

Cuphat committed:
Version Refactor

Reworks version checks to make it easier for branches to have their own individual version tracking. Version checks are performed using the branch URL specified, and now checks the supplementary version as well.

The version bytes injected into the ROM header at address 0x35 has been expanded to 5 bytes. In addition to the 3 base version bytes, it now has branch identifier for the fourth byte and the supplementary version for the fifth byte.

Roman971 commented:
Address 0x38 in the header is reserved and documented as a word ie. 4 bytes ([it's even read as a word in oot, tho it's ultimately unused](https://github.com/zeldaret/oot/blob/82e04ede5f3fef20fb95f6a16b88b8a684f49954/src/boot/z_locale.c#L8)), so I don't think we should write anything to it. In particular, it could cause issues since emulators may read it as a word and change their behavior based on that.
If we're looking for free space in the header, the 8 bytes at 0x18 are unused afaik, so we could probably just use that.

Cuphat committed:
Fallback to base_version when full_version isn't found.

Cuphat committed:
Version Bytes Update

Branch Identifier and Supplementary version are no longer at 0x38 and 0x39.

A second copy of the version bytes, with branch identifier and supplementary version information, is now at 0x19-0x1D.

Cuphat commented:
While I think it'd _probably_ be alright (seemed to work fine on EverDrive/Wii VC/BizHawk/PJ64, I didn't test RetroArch), the fact that it is explicitly used as a 4-byte in the code convinced me to leave 0x38/0x39 alone, so now it puts the "full" version bytes on 0x19-0x1D.
